app/src/main/kotlin/com/intive/picover/images/view/ImagesScreen.kt

```diff
@@ -0,0 +1,75 @@
+package com.intive.picover.images.view
+
+import android.net.Uri
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.lazy.items
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material3.Card
+import androidx.compose.material3.CardDefaults
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.unit.dp
+import com.intive.picover.common.loader.PicoverLoader
+import com.intive.picover.common.platform.viewmodel.ImagesViewModel
+import com.intive.picover.common.platform.viewmodel.LatestFirebaseStorageState.Error
+import com.intive.picover.common.platform.viewmodel.LatestFirebaseStorageState.Loading
+import com.intive.picover.common.platform.viewmodel.LatestFirebaseStorageState.Success
+import com.skydoves.landscapist.coil.CoilImage
+
+@Composable
+fun ImagesScreen(viewModel: ImagesViewModel) {
+	val state by viewModel.uiState.collectAsState()
+
+	when (state) {
+		is Success -> {
+			val uris = (state as Success).uris
+			LazyColumn(
```

Comment:
I would extract it to separate function to keep the main function as simple as possible (to make the reader focus on the when statement)